core/src/main/java/org/apache/brooklyn/core/objs/AbstractBrooklynObject.java

```diff
@@ -83,11 +90,18 @@ public AbstractBrooklynObject(Map<?, ?> properties) {
         // correct behaviour should be to inherit context's search path, perhaps, though maybe that's better done as spec?
         // in any case, should not define it as _the_ catalog item ID; also see assignment based on parent
         // in CatalogUtils.setCatalogItemIdOnAddition
-        catalogItemId = ApiObjectsFactory.get().getCatalogItemIdFromContext();
+        setCatalogItemId(ApiObjectsFactory.get().getCatalogItemIdFromContext());
```

Comment:
context ID should be added to the _search path_ not used to set the ID of the item itself -- this was the problem spot that prompted keeping the IDs separate

and can remove the TODO now!  :)